src/chain_sync/chain_muxer.rs

```diff
@@ -165,6 +165,11 @@ pub struct ChainMuxer<DB, M> {
 
     /// When `stateless_mode` is true, forest connects to the P2P network but does not sync to HEAD.
     stateless_mode: bool,
+
+    /// `true` if `ChainMuxer` is in "follow" mode.
+    // NOTE: This is a temporary workaround for `Filecoin.SyncSubmitBlock` as it can only work on
+    // "follow".
+    is_follow: Arc<Mutex<bool>>,
```

Comment:
You can derive this from the `worker_state`.

Reply:
Ah, `Complete`. Sounds much better! Cheers.